### ISSUE TITLE
Compilation fails with Boost 1.66

### ISSUE THREAD

**Issue by BenjaminRueth** [precice/precice#93]
Compilation fails due to wrong number of template arguments.

See for example precice/src/com/SocketCommunication.hpp:262

```
typedef boost::asio::basic_stream_socket<TCP, SocketService> Socket;
```

* two arguments are accepted with Boost 1.62 (http://www.boost.org/doc/libs/1_62_0/doc/html/boost_asio/reference/basic_stream_socket.html)
* only one argument are accepted with Boost 1.66 (http://www.boost.org/doc/libs/1_66_0/doc/html/boost_asio/reference/basic_stream_socket.html)

**Comment by floli** [comment]
Should already be fixed in a41d9f4c711e9cfafc5ce21ed9062e0563f916af

**Comment by richahert** [comment]
It is fixed, I successfully built it with boost 1.66
